<a href="https://colab.research.google.com/github/atlas-github/fi_analytics/blob/master/Chapter_11_Hypothesis_testing_concepts_and_examples.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##One sample t-test

Using a short Python script, the code would be:

In [1]:
from scipy import stats

sample1 = [128, 126, 119, 114, 146, 140, 135, 138, 135, 128, 114, 129, 150, 124, 142, 118, 134, 126, 127, 127]

stats.ttest_1samp(sample1, 120)

Ttest_1sampResult(statistic=4.469784055833502, pvalue=0.0002625920272762138)

The p-value is significantly less than 0.05, so the null hypothesis is rejected. There is a statistically significant difference between the blood pressures of the sample and the population

##Two sample t-test

The Python script would look like this:

In [2]:
from scipy import stats

sample1 = [128, 126, 119, 114, 146, 140, 135, 138, 135, 128, 114, 129, 150, 124, 142, 118, 134, 126, 127, 127]

sample2 = [118, 116, 111, 121, 122, 132, 121, 112, 117, 124, 120, 128, 128, 128, 109, 118, 121, 123, 122, 129]

stats.ttest_ind(sample1, sample2, equal_var=False)

Ttest_indResult(statistic=3.418427884676446, pvalue=0.0017469015807571465)

##Paired t-test

The Python script would look like this:

In [4]:
from scipy import stats

before = [9, 6.1, 8.5, 4.7, 6.2, 6.4, 5.7, 10.1, 5.3, 2.8]
after = [10.9, 6.9, 9.6, 4.8, 6.1, 10.9, 11.5, 11.7, 9.9, 6.2]

stats.ttest_rel(before, after)

Ttest_relResult(statistic=-3.624485995178213, pvalue=0.005532940816100147)

The p-value obtained is less than 0.05, so the null hypothesis should be rejected. 

##z-test

Using the [statsmodels](https://www.statsmodels.org/stable/generated/statsmodels.stats.weightstats.ztest.html) library for the z-test, the script should look like below.  

In [5]:
from statsmodels.stats.weightstats import ztest

gain = [3.50, 2.38, 3.33, 1.92, 2.47, 2.48, 2.35, 3.72, 2.22, 1.28,
        4.03, 2.65, 3.60, 1.95, 2.43, 3.98, 4.23, 4.25, 3.70, 2.42,
        3.40, 2.38, 3.23, 2.48, 2.30, 3.72, 4.03, 2.70, 3.55, 2.43,
        4.03, 4.23, 3.35, 2.43, 3.18]

ztest(x1 = gain, value = 2.5, alternative = 'larger')

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


(4.006354707207111, 3.083151329152138e-05)

##Chi-square test

Using the scipy library, the script would resemble the following:

In [6]:
from scipy.stats import chisquare
chisquare([450, 350, 100, 350, 450, 300])

Power_divergenceResult(statistic=250.00000000000003, pvalue=5.4969224674550846e-52)

The `chisquare` function returns two values: the chi squared test `statistic` which is 250, and the `p-value` of the test. Since the `p-value` is lower than 0.05, this indicates there is some association between the variables. This is expected, as Party C seems to have a larger pull factor among women than men.